From 5f85b3532f8a9811e8b24f7d4529dedd9eafbd58 Mon Sep 17 00:00:00 2001
From: Nanley Chery <nanley.g.chery@intel.com>
Date: Thu, 17 Feb 2022 16:20:00 -0800
Subject: [PATCH 03/25] anv: Add a perf warning in anv_BindImageMemory2

It reports: "BO lacks implicit CCS. Disabling the CCS aux usage."

Reviewed-by: Lionel Landwerlin <lionel.g.landwerlin@intel.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/15068>
(cherry picked from commit 203c8be09f7894600add91109060ed44f92702b0)
---
 src/intel/vulkan/anv_image.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/intel/vulkan/anv_image.c b/src/intel/vulkan/anv_image.c
index 93975a06749..a5948af386f 100644
--- a/src/intel/vulkan/anv_image.c
+++ b/src/intel/vulkan/anv_image.c
@@ -1843,6 +1843,9 @@ VkResult anv_BindImageMemory2(
          if (!isl_aux_usage_has_ccs(image->planes[p].aux_usage))
             continue;
 
+         anv_perf_warn(VK_LOG_OBJS(&image->vk.base),
+                       "BO lacks implicit CCS. Disabling the CCS aux usage.");
+
          if (image->planes[p].aux_surface.memory_range.size > 0) {
             assert(image->planes[p].aux_usage == ISL_AUX_USAGE_HIZ_CCS ||
                    image->planes[p].aux_usage == ISL_AUX_USAGE_HIZ_CCS_WT);
-- 
2.34.1

